File: club_management/members/report/pagos_del_dia/pagos_del_dia.py
# Copyright (c) 2026, fblasco1 and contributors
# For license information, please see license.txt

from __future__ import annotations

from typing import Any

from club_management.members.services.recaudacion_por_concepto import (
	get_recaudacion_unificada_report_columns,
	get_recaudacion_unificada_report_data,
	get_recaudacion_unificada_report_summary,
)


def execute(
	filters: dict[str, Any] | None = None,
) -> tuple[
	list[dict[str, Any]],
	list[dict[str, Any]],
	None,
	None,
	list[dict[str, Any]],
]:
	"""Legado: misma vista que Cobranza por fechas con un solo día."""
	raw = dict(filters or {})
	fecha = raw.get("fecha") or raw.get("fecha_desde")
	merged = {
		**raw,
		"fecha_desde": fecha,
		"fecha_hasta": fecha,
	}
	return (
		get_recaudacion_unificada_report_columns(merged),
		get_recaudacion_unificada_report_data(merged),
		None,
		None,
		get_recaudacion_unificada_report_summary(merged),
	)

File: club_management/members/report/pagos_del_dia/pagos_del_dia.js
// Copyright (c) 2026, fblasco1 and contributors
// For license information, please see license.txt

frappe.query_reports["Pagos del dia"] = {
	filters: [
		{
			fieldname: "fecha",
			label: __("Fecha"),
			fieldtype: "Date",
			default: frappe.datetime.get_today(),
			reqd: 1,
		},
	],
	onload(report) {
		// Redirige al informe unificado (mismo día en Desde/Hasta)
		const fecha = report.get_filter_value("fecha") || frappe.datetime.get_today();
		frappe.set_route("query-report", "Recaudacion por concepto", {
			fecha_desde: fecha,
			fecha_hasta: fecha,
		});
	},
};
